### PR TITLE
refactor: refactor CollectionPagination

### PR DESCRIPTION
**Pull request by debitan**
The component can receive minimum number of props and infer the others it needs. 

Tests  updated to match. 

Logic for truncation is now within the component itself. This logic is largely based on the existing logic in the frontend repo. Welcome suggestions on how to simplify it. 

Screenshot 2019-10-16 at 13 25 22

### CODE REVIEW

File: src/collection-list/CollectionList.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        <CollectionItem
+          id={profile.id}
+          headingUrl={profile.headerLink}
+          headingText={profile.headerText}
+          badges={profile.badges}
+          metadata={profile.metadata}
+        />
+      ))}
+      <CollectionPagination
+        totalPages={Math.floor(totalItems / pageLimit) + 1}
```
Comment (ian-leggett) [inline src/collection-list/CollectionList.jsx]: I would probably move this up so the expression isn't in the prop value.

File: src/collection-list/CollectionList.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        totalItems={totalItems}
+        itemName={itemName}
+        addItemUrl={addItemUrl}
+      />
+      <CollectionDownload
+        totalItems={totalItems}
+        itemName={itemName}
+        downloadUrl={downloadUrl}
+      />
+      {profiles.map((profile) => (
```
Comment (ian-leggett) [inline src/collection-list/CollectionList.jsx]: Maybe deconstruct here, for example `{id, headerLink, headerText, badges, metadata}`

**Comment by debitan** [comment]
The code coverage is dropping here because of the truncation function. I have written a specific test for it, but it still doesn't fully satisfy codecov. However, as discussed at the FE architecture meeting this morning, I have already tested the component itself, so testing the function within it doesn't provide any value anyway. So with permission I would force the merger of this PR to override the codecov block.

File: src/collection-list/CollectionPagination.jsx [hunk 3]
```diff
@@ -75,16 +76,31 @@ const StyledSpan = styled('span')`
...
   totalPages,
-  currentPage,
   previous,
   next,
-  pages,
+  apiEndpoint,
+  pageLimit,
 }) {
+  const currentPage = next
+    ? Number(next.split('&offset=')[1]) / pageLimit
```
[the remark this answers is not in the document]
Comment (web-bert) [reply src/collection-list/CollectionPagination.jsx]: You can pass in the separator, default to `&` and then pass in the `?` for the first one?

[the remark this answers is not in the document]
Comment (web-bert) [reply src/collection-list/CollectionPagination.jsx]: i.e.
```javascript
getParam('offset')
```
```javascript
getParam('limit', '?')
```

File: src/collection-list/__tests__/truncatePages.test.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+  describe('when 4 pages are passed', () => {
+    beforeAll(() => {
+      const currentPage = 1
+      const untruncatedPages = fourPages
+      pages = truncatePages(untruncatedPages, currentPage)
+    })
+
+    test('truncation should return the same object if < 5 pages', () => {
+      expect(pages).toEqual(pages)
```
[the remark this answers is not in the document]
Comment (debitan) [reply src/collection-list/__tests__/truncatePages.test.js]: it should have been, but I'm now testing this function in the component test file.

**Pull request merged**